src/test/java/htsjdk/samtools/cram/structure/SliceTests.java

```diff
@@ -72,238 +75,180 @@ public void testFailsMD5Check() throws IOException {
+                },
+                {
+                        getNoRefRecords(),
+                        true,
+                        TEST_RECORD_COUNT, SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX, Slice.NO_ALIGNMENT_START, Slice.NO_ALIGNMENT_SPAN
+                },
+
+                // show that not having a valid alignment start does nothing
+                // in the Coordinate-Sorted case (so it remains a Multi-Ref Slice)
+                // but causes the reads to be marked as Unplaced otherwise (so it becomes an Unmapped Slice)
+
+                {
+                        getNoStartRecords(),
+                        true,
+                        TEST_RECORD_COUNT, Slice.MULTI_REFERENCE, Slice.NO_ALIGNMENT_START, Slice.NO_ALIGNMENT_SPAN
+                },
+                {
+                        getNoStartRecords(),
+                        false,
+                        TEST_RECORD_COUNT, SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX, Slice.NO_ALIGNMENT_START, Slice.NO_ALIGNMENT_SPAN
+                },
+
+        };
     }
 
-    @Test
-    public void testMultiRefBuild() {
-        final List<CramCompressionRecord> records = new ArrayList<>();
-        for (int i = 0; i < 10; i++) {
-            final CramCompressionRecord record = new CramCompressionRecord();
-            record.readBases = "AAA".getBytes();
-            record.qualityScores = "!!!".getBytes();
-            record.readLength = 3;
-            record.readName = "" + i;
-            record.sequenceId = i;
-            record.alignmentStart = i + 1;
-            record.setLastSegment(true);
-            record.readFeatures = Collections.emptyList();
-
-            if (i % 2 == 0) {
-                record.setSegmentUnmapped(true);
-            } else {
-                record.setSegmentUnmapped(false);
-            }
-
-            records.add(record);
-        }
-
-        final CompressionHeader header = new CompressionHeaderFactory().build(records, null, true);
-
+    @Test(dataProvider = "forBuildTests")
+    public void testBuild(final List<CramCompressionRecord> records,
+                          final boolean coordinateSorted,
+                          final int expectedRecordCount,
+                          final int expectedSequenceId,
+                          final int expectedAlignmentStart,
+                          final int expectedAlignmentSpan) {
+        final CompressionHeader header = new CompressionHeaderFactory().build(records, null, coordinateSorted);
         final Slice slice = Slice.buildSlice(records, header);
-
-        Assert.assertEquals(slice.nofRecords, 10);
-        Assert.assertEquals(slice.sequenceId, Slice.MULTI_REFERENCE);
-        Assert.assertEquals(slice.alignmentStart, Slice.NO_ALIGNMENT_START);
-        Assert.assertEquals(slice.alignmentSpan, Slice.NO_ALIGNMENT_SPAN);
+        assertSliceState(slice, expectedRecordCount, expectedSequenceId, expectedAlignmentStart, expectedAlignmentSpan);
```

Comment:
I like the way this test/provider combination is written - simple and declarative - thx.